### Source:
- Betancourt L, Pico S, Rojas E, Gordo MJ, Oliva JC, Almirall J, et al. Relationship between intraperitoneal pressure and the development of hernias in peritoneal dialysis: confirmation for the first time of a widely accepted concept. International Urology and Nephrology. 2023; https://doi.org/10.1007/s11255-023-03663-5.

The odds ratio is calculated as:

$$
OR = \frac{a \times d}{b \times c}
$$

Given the data provided:

1. The overall mean IPP supine position for the entire cohort was 16.6.
2. From the provided data, 23 patients had hernias.

In [3]:
from scipy.stats import norm
# Given data
mean_with_hernias = 18.4
std_dev_with_hernias = 4.58
mean_without_hernias = 16.2
std_dev_without_hernias = 4.52

# Total number of patients with and without hernias
patients_with_hernias = 23
patients_without_hernias = 101

# Threshold for high IPP supine position
threshold = 16.6

# Estimate the number of patients with "high" IPP supine position in each group
high_IPP_with_hernias = patients_with_hernias * (1 - norm.cdf((threshold - mean_with_hernias) / std_dev_with_hernias))
high_IPP_without_hernias = patients_without_hernias * (1 - norm.cdf((threshold - mean_without_hernias) / std_dev_without_hernias))

# Estimate the number of patients with "low" IPP supine position in each group
low_IPP_with_hernias = patients_with_hernias - high_IPP_with_hernias
low_IPP_without_hernias = patients_without_hernias - high_IPP_without_hernias

# Create the 2x2 contingency table
contingency_table = {
    '': ['High IPP supine position', 'Low IPP supine position'],
    'Hernias': [high_IPP_with_hernias, low_IPP_with_hernias],
    'No Hernias': [high_IPP_without_hernias, low_IPP_without_hernias]
}

display(contingency_table)

# Calculate odds ratio
OR = (high_IPP_with_hernias * low_IPP_without_hernias) / (low_IPP_with_hernias * high_IPP_without_hernias)
OR


{'': ['High IPP supine position', 'Low IPP supine position'],
 'Hernias': [15.015437235671369, 7.984562764328631],
 'No Hernias': [46.93888145074807, 54.06111854925193]}

2.1659036585477387

The estimated odds ratio (OR) for developing hernias based on the IPP supine position, using the threshold of 16.6, is approximately $2.17$. 